src/Data/Bin/Base.agda

```diff
@@ -0,0 +1,114 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Arithmetic of binary represented natural numbers. Initial part.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Bin.Base where
+
+open import Algebra.FunctionProperties using (Op₂)
+open import Data.Nat        using (ℕ) renaming (suc to 1+_; _*_ to _*ₙ_)
+open import Function        using (_∘_)
+open import Relation.Binary using (Decidable)
+open import Relation.Binary.PropositionalEquality using (_≡_; refl; cong)
+open import Relation.Nullary                      using (yes; no)
+
+
+data Bin : Set
+  where
+  0#    : Bin
+  2suc  : Bin → Bin    -- \n→ 2*(1+n)  arbitrary nonzero even
+  suc2* : Bin → Bin    -- \n→ 1 + 2n   arbitrary odd
+--
+-- This representation is unique for each natural number.
+
+
+size : Bin → ℕ       -- (number of constructors) - 1.
```

Comment:
``size (x : Bin)``  is actually  (numberOfConstructorsIn x) - 1.
``size 0# = 0``, while integer part of ``(log₂ 0)`` is ``(- infinity)``, that is it does not exist in integer numbers.   
Further,
```
size 2B = 1 = [log₂ 2],
size 3B = 2;               [log₂ 3] = 1.
```
I think, they are close to each other, but they have different definition domains, and it is better to differ ``size`` from ``log₂``.

``log₂`` is implemented in BFLib-0.1, Bin/Logarithm.agda, it can be included later if you do not care for the source size.

Reply:
Should this function be called length? It is effectively the lenght of the list
of `2*1+` and `1+2*` constructors tacked on top of `0#`.